internal/app/init/pkg/system/services/kubeadm/kubeadm.go

```diff
@@ -228,41 +228,43 @@ func CreateTrustdClients(data *userdata.UserData) ([]proto.TrustdClient, error)
 func Download(ctx context.Context, client proto.TrustdClient, file *proto.ReadFileRequest, content chan<- []byte) {
 	select {
 	case <-ctx.Done():
-		return
 	case content <- download(ctx, client, file):
-		ctx.Done()
-		return
 	}
 }
 
 func download(ctx context.Context, client proto.TrustdClient, file *proto.ReadFileRequest) []byte {
-	var resp *proto.ReadFileResponse
-	var err error
-
-	attempt := 0
+	var (
+		resp            *proto.ReadFileResponse
+		err             error
+		attempt, snooze float64
+	)
 	maxWait := float64(64)
 
 	for {
-		if ctx.Err() != nil {
-			return []byte{}
+		ctxTimeout, ctxTimeoutCancel := context.WithTimeout(ctx, 2*time.Second)
+		defer ctxTimeoutCancel()
```

Comment:
I think these 2 lines should be moved up and out of the for loop.

Reply:
I agree we can't do `defer` in `for` as they would all stacked up, but if we move it out of `for`, we would enforce 2 sec timeout on the whole function, not each attempt.

The easiest way is to extract for loop body into a function:

```
for {
   func() {
     ctxTimeout, ... 
     defer ctxTimeoutCancel()

     ....
   }()
}
```

Reply:
hmm ... maybe not ... the concern is the stacking of `deferred cancels